---
title: "CLIP-Seq Workflow Template"
author: "Author: Daniela Cassol (danielac@ucr.edu) and Thomas Girke (thomas.girke@ucr.edu)"
date: "Last update: `r format(Sys.time(), '%d %B, %Y')`"
output:
  BiocStyle::html_document:
    toc_float: true
    code_folding: show
  BiocStyle::pdf_document: default
package: systemPipeR
vignette: |
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{Clip_seq Workflow}
  %\VignetteEngine{knitr::rmarkdown}
fontsize: 14pt
bibliography: bibtex.bib
---


```{css, echo=FALSE}
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
```

<!-- 
Compile from command-line  
Rscript -e "rmarkdown::render('systemPipeChIPseq.Rmd', c('BiocStyle::html_document'), clean=F); knitr::knit('systemPipeChIPseq.Rmd', tangle=TRUE)"; Rscript ../md2jekyll.R systemPipeChIPseq.knit.md 13; Rscript -e "rmarkdown::render('systemPipeChIPseq.Rmd', c('BiocStyle::pdf_document'))"
-->


```{r style, echo = FALSE, results = 'asis'}
BiocStyle::markdown()
options(width=100, max.print=1000)
knitr::opts_chunk$set(
    eval=as.logical(Sys.getenv("KNITR_EVAL", "TRUE")),
    cache=as.logical(Sys.getenv("KNITR_CACHE", "TRUE")))
```

```{r setup, echo=FALSE, message=FALSE, warning=FALSE}
suppressPackageStartupMessages({
    library(systemPipeR)
    library(BiocParallel)
    library(Biostrings)
    library(Rsamtools)
    library(GenomicRanges)
    library(ggplot2)
    library(GenomicAlignments)
    library(ShortRead)
    library(ape)
    library(batchtools)
})
```

# Introduction

Users want to provide here background information about the CLIP-Seq project.

## Background and objectives

This report describes the analysis of several e-CLIP experiments studying the RNA biding proteins.

## Experimental design

Typically, users want to specify here all information relevant for the
analysis of their NGS study. This includes detailed descriptions of
FASTQ files, experimental design, reference genome, gene annotations,
etc.

# Workflow environment

## Generate workflow environment

## Run workflow

### Run R session on computer node

The `systemPipeR` package needs to be loaded to perform the analysis steps shown in this report [@H_Backman2016-bt].

```{r load_systempiper, eval=FALSE}
library(systemPipeR)
```

# Read preprocessing

## Experiment definition provided by `targets` file

The `targets` file defines all FASTQ files and sample comparisons of the analysis workflow.

```{r load_targets_file, eval=TRUE}
targetspath <- system.file("extdata", "targets.txt", package="systemPipeR")
targets <- read.delim(targetspath, comment.char = "#")
targets
```

## FASTQ quality report 

The following `seeFastq` and `seeFastqPlot` functions generate and plot a series of useful quality
statistics for a set of FASTQ files including per cycle quality box
plots, base proportions, base-level quality trends, relative k-mer
diversity, length and occurrence distribution of reads, number of reads
above quality cutoffs and mean quality distribution. The results are
written to a PDF file named `fastqReport.pdf`.

```{r fastq_report, eval=FALSE}
args <- systemArgs(sysma="param/trim_clipper.param", mytargets="targets.txt")
fqlist <- seeFastq(fastq=infile1(args), batchsize=100000, klength=8)
pdf("./results/fastqReport.pdf", height=18, width=4*length(fqlist))
seeFastqPlot(fqlist)
dev.off()
```

![](results/fastqReport.png)
<div align="center">Figure 1: FASTQ quality report for 2 samples</div></br>

## Read quality filtering and trimming

### Remove Adaptor with fastx_clipper 

Removing Adapter sequence using fastx_toolkit.
OBS: The adapter at the 3’ end should be AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC and the 10nt on the 5’ end of the read are the barcode.

```{r fastx_clipper, eval=FALSE}
args <- systemArgs(sysma="param/trim_clipper.param", mytargets="targets.txt")
runCommandline(args)
writeTargetsout(x=args, file="targets_clip.txt", overwrite=TRUE)
```

#### First look at the data

```{r seeDistribution, eval=FALSE}
source("seeDistribution.R")
pdf("results/Distribution_AdapterRemoval.pdf")
seeDistribution(outfile1(args), file="fastq", batchsize = 100000)
dev.off()
```

## Mapping ALL reads

In order to have a better understanding of the data, it's possible to map all the reads with the reference genome.

### Remove barcode

Second step, remove barcode. In this data, N = 10.

```{r fastx_triming_barcode, preprocess_reads1, eval=FALSE}
args <- systemArgs(sysma="param/trim.param", mytargets="targets_clip.txt")
runCommandline(args)
```

#### Reads Length Distribution 

Read length distribution after barcode removal.

```{r seeDistribution_afterBarcodeRemoval, eval=FALSE}
source("seeDistribution.R")
pdf("results/Distribution_barcodeRemoval.pdf")
seeDistribution(outfile1(args), file="fastq", batchsize = 100000)
dev.off()
```

### Read mapping with `HISAT2`

```{r hisat_alignment21, eval=FALSE}
args <- systemArgs(sysma="param/hisat2.param", mytargets="targets_clip_trim.txt")
sysargs(args)[1] # Command-line parameters for first FASTQ file
moduleload(modules(args))
#system("hisat2-build ./data/dataH/hg19.fa ./data/dataH/hg19.fa")
resources <- list(walltime=1440, ntasks=1, ncpus=12, memory=10240, email="danielac@ucr.edu")
hisat2 <- clusterRun(args, conffile=".batchtools.conf.R", template="batchtools.slurm.tmpl", runid="01", Njobs=12, resourceList=resources)
getStatus(reg=hisat2)
```

Check whether all BAM files have been created

```{r check_files_exist, eval=FALSE}
file.exists(outpaths(args))
```

Check all reads mapping results.

```{r alignStatus_READS2, eval=FALSE}
read_statsDF_READS <- alignStats(args=args)
write.table(read_statsDF_READS, "results/alignStats_READS.xls", row.names=FALSE, quote=FALSE, sep="\t")
```

## Mapping Unique reads

### Collapsing identical sequences 

Collapsing identical sequences in a FASTQ/A file into a single sequence (while maintaining reads counts).

```{r fastx_collapse, eval=FALSE}
args <- systemArgs(sysma="param/trim_collapser.param", mytargets="targets_clip.txt")
sysargs(args)[1]
runCommandline(args, make_bam=FALSE)
writeTargetsout(x=args, file="targets_clip_collapser.txt", overwrite=TRUE)
```

#### Reads Length Distribution 

Read length distribution after collapsing the identical reads.

```{r seeDistribution_afterCollapsing, eval=FALSE}
source("seeDistribution.R")
pdf("results/Distribution_collapsing.pdf")
seeDistribution(outfile1(args), file="fasta")
dev.off()
```

### Removing Barcode

Remove barcode after collapse.

```{r fastx_triming_barcode1, preprocess_reads, eval=FALSE}
args <- systemArgs(sysma="param/trim_fasta.param", mytargets="targets_clip_collapser.txt")
runCommandline(args, make_bam=FALSE)
writeTargetsout(x=args, file="targets_clip_collap_trim.txt", overwrite=TRUE)
```

#### Reads Length Distribution 

Read length distribution after barcode removal.

```{r seeDistribution_afterCollapsingandtrim, eval=FALSE}
source("seeDistribution.R")
pdf("results/Distribution_Collapsing_barcodeRemoval.pdf")
seeDistribution(outfile1(args), file="fasta")
dev.off()
```

### Read mapping with `HISAT2`

```{r hisat_alignment2, eval=FALSE}
args <- systemArgs(sysma="param/hisat2_unique.param", mytargets="targets_clip_collap_trim.txt")
sysargs(args)[1] # Command-line parameters for first FASTQ file
moduleload(modules(args))
#system("hisat2-build ./data/dataH/hg19.fa ./data/dataH/hg19.fa")
resources <- list(walltime=1440, ntasks=1, ncpus=12, memory=10240, email="danielac@ucr.edu")
hisat2 <- clusterRun(args, conffile=".batchtools.conf.R", template="batchtools.slurm.tmpl", runid="01", Njobs=12, resourceList=resources)
getStatus(reg=hisat2)
```

Check whether all BAM files have been created

```{r check_files_exist1, eval=FALSE}
file.exists(outpaths(args))
```
Check all reads mapping results.

```{r alignStatus_READS1, eval=FALSE}
read_statsDF_READS <- alignStats(args=args)
write.table(read_statsDF_READS, "results/alignStats_READS_unique.xls", row.names=FALSE, quote=FALSE, sep="\t")
```


# Version Information

```{r sessionInfo}
sessionInfo()
```

# Funding

This project was supported by funds from the National Science Foundation (NSF).

# References

