<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SPclipseq Workflow Template • SPclipseq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="SPclipseq Workflow Template">
<meta property="og:description" content="SPclipseq">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-CCMXLYXWYV"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-CCMXLYXWYV');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPclipseq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/SPclipseq.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/systemPipeR/SPclipseq/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="SPclipseq_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>SPclipseq Workflow Template</h1>
                        <h4 class="author">Author: Daniela Cassol (<a href="mailto:danielac@ucr.edu" class="email">danielac@ucr.edu</a>) and Thomas Girke (<a href="mailto:thomas.girke@ucr.edu" class="email">thomas.girke@ucr.edu</a>)</h4>
            
            <h4 class="date">Last update: 23 April, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/systemPipeR/SPclipseq/blob/master/vignettes/SPclipseq.Rmd"><code>vignettes/SPclipseq.Rmd</code></a></small>
      <div class="hidden name"><code>SPclipseq.Rmd</code></div>

    </div>

    
    
<style type="text/css">
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
</style>
<!-- 
Compile from command-line  
Rscript -e "rmarkdown::render('SPclipseq.Rmd', c('BiocStyle::html_document'), clean=TRUE)"
--><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Users want to provide here background information about the CLIP-Seq project.</p>
<div id="background-and-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#background-and-objectives" class="anchor"></a>Background and objectives</h2>
<p>This report describes the analysis of several e-CLIP experiments studying the RNA biding proteins.</p>
</div>
<div id="experimental-design" class="section level2">
<h2 class="hasAnchor">
<a href="#experimental-design" class="anchor"></a>Experimental design</h2>
<p>Typically, users want to specify here all information relevant for the analysis of their NGS study. This includes detailed descriptions of FASTQ files, experimental design, reference genome, gene annotations, etc.</p>
</div>
</div>
<div id="workflow-environment" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow-environment" class="anchor"></a>Workflow environment</h1>
<div id="generate-workflow-environment" class="section level2">
<h2 class="hasAnchor">
<a href="#generate-workflow-environment" class="anchor"></a>Generate workflow environment</h2>
</div>
<div id="run-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#run-workflow" class="anchor"></a>Run workflow</h2>
<div id="run-r-session-on-computer-node" class="section level3">
<h3 class="hasAnchor">
<a href="#run-r-session-on-computer-node" class="anchor"></a>Run R session on computer node</h3>
<p>The <code>systemPipeR</code> package needs to be loaded to perform the analysis steps shown in this report <span class="citation">(H Backman and Girke 2016)</span>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://systempipe.org/">systemPipeR</a></span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
<div id="read-preprocessing" class="section level1">
<h1 class="hasAnchor">
<a href="#read-preprocessing" class="anchor"></a>Read preprocessing</h1>
<div id="experiment-definition-provided-by-targets-file" class="section level2">
<h2 class="hasAnchor">
<a href="#experiment-definition-provided-by-targets-file" class="anchor"></a>Experiment definition provided by <code>targets</code> file</h2>
<p>The <code>targets</code> file defines all FASTQ files and sample comparisons of the analysis workflow.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">targetspath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"targets.txt"</span>, package<span class="op">=</span><span class="st">"systemPipeR"</span><span class="op">)</span>
<span class="va">targets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="va">targetspath</span>, comment.char <span class="op">=</span> <span class="st">"#"</span><span class="op">)</span>
<span class="va">targets</span></code></pre></div>
<pre><code>##                       FileName SampleName Factor SampleLong Experiment        Date
## 1  ./data/SRR446027_1.fastq.gz        M1A     M1  Mock.1h.A          1 23-Mar-2012
## 2  ./data/SRR446028_1.fastq.gz        M1B     M1  Mock.1h.B          1 23-Mar-2012
## 3  ./data/SRR446029_1.fastq.gz        A1A     A1   Avr.1h.A          1 23-Mar-2012
## 4  ./data/SRR446030_1.fastq.gz        A1B     A1   Avr.1h.B          1 23-Mar-2012
## 5  ./data/SRR446031_1.fastq.gz        V1A     V1   Vir.1h.A          1 23-Mar-2012
## 6  ./data/SRR446032_1.fastq.gz        V1B     V1   Vir.1h.B          1 23-Mar-2012
## 7  ./data/SRR446033_1.fastq.gz        M6A     M6  Mock.6h.A          1 23-Mar-2012
## 8  ./data/SRR446034_1.fastq.gz        M6B     M6  Mock.6h.B          1 23-Mar-2012
## 9  ./data/SRR446035_1.fastq.gz        A6A     A6   Avr.6h.A          1 23-Mar-2012
## 10 ./data/SRR446036_1.fastq.gz        A6B     A6   Avr.6h.B          1 23-Mar-2012
## 11 ./data/SRR446037_1.fastq.gz        V6A     V6   Vir.6h.A          1 23-Mar-2012
## 12 ./data/SRR446038_1.fastq.gz        V6B     V6   Vir.6h.B          1 23-Mar-2012
## 13 ./data/SRR446039_1.fastq.gz       M12A    M12 Mock.12h.A          1 23-Mar-2012
## 14 ./data/SRR446040_1.fastq.gz       M12B    M12 Mock.12h.B          1 23-Mar-2012
## 15 ./data/SRR446041_1.fastq.gz       A12A    A12  Avr.12h.A          1 23-Mar-2012
## 16 ./data/SRR446042_1.fastq.gz       A12B    A12  Avr.12h.B          1 23-Mar-2012
## 17 ./data/SRR446043_1.fastq.gz       V12A    V12  Vir.12h.A          1 23-Mar-2012
## 18 ./data/SRR446044_1.fastq.gz       V12B    V12  Vir.12h.B          1 23-Mar-2012</code></pre>
</div>
<div id="fastq-quality-report" class="section level2">
<h2 class="hasAnchor">
<a href="#fastq-quality-report" class="anchor"></a>FASTQ quality report</h2>
<p>The following <code>seeFastq</code> and <code>seeFastqPlot</code> functions generate and plot a series of useful quality statistics for a set of FASTQ files including per cycle quality box plots, base proportions, base-level quality trends, relative k-mer diversity, length and occurrence distribution of reads, number of reads above quality cutoffs and mean quality distribution. The results are written to a PDF file named <code>fastqReport.pdf</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">args</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span><span class="op">(</span>sysma<span class="op">=</span><span class="st">"param/trim_clipper.param"</span>, mytargets<span class="op">=</span><span class="st">"targets.txt"</span><span class="op">)</span>
<span class="va">fqlist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/seeFastq.html">seeFastq</a></span><span class="op">(</span>fastq<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">infile1</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span>, batchsize<span class="op">=</span><span class="fl">100000</span>, klength<span class="op">=</span><span class="fl">8</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span><span class="op">(</span><span class="st">"./results/fastqReport.pdf"</span>, height<span class="op">=</span><span class="fl">18</span>, width<span class="op">=</span><span class="fl">4</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fqlist</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/seeFastq.html">seeFastqPlot</a></span><span class="op">(</span><span class="va">fqlist</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<img src="results/fastqReport.png"><div align="center">
Figure 1: FASTQ quality report for 2 samples
</div>
<p></p>
</div>
<div id="read-quality-filtering-and-trimming" class="section level2">
<h2 class="hasAnchor">
<a href="#read-quality-filtering-and-trimming" class="anchor"></a>Read quality filtering and trimming</h2>
<div id="remove-adaptor-with-fastx_clipper" class="section level3">
<h3 class="hasAnchor">
<a href="#remove-adaptor-with-fastx_clipper" class="anchor"></a>Remove Adaptor with fastx_clipper</h3>
<p>Removing Adapter sequence using fastx_toolkit. OBS: The adapter at the 3’ end should be <code>AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC</code> and the 10nt on the 5’ end of the read are the barcode.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">args</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span><span class="op">(</span>sysma<span class="op">=</span><span class="st">"param/trim_clipper.param"</span>, mytargets<span class="op">=</span><span class="st">"targets.txt"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">args</span>, file<span class="op">=</span><span class="st">"targets_clip.txt"</span>, overwrite<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="first-look-at-the-data" class="section level4">
<h4 class="hasAnchor">
<a href="#first-look-at-the-data" class="anchor"></a>First look at the data</h4>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="st">"seeDistribution.R"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span><span class="op">(</span><span class="st">"results/Distribution_AdapterRemoval.pdf"</span><span class="op">)</span>
<span class="fu">seeDistribution</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outfile1</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span>, file<span class="op">=</span><span class="st">"fastq"</span>, batchsize <span class="op">=</span> <span class="fl">100000</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
<div id="mapping-all-reads" class="section level2">
<h2 class="hasAnchor">
<a href="#mapping-all-reads" class="anchor"></a>Mapping ALL reads</h2>
<p>In order to have a better understanding of the data, it’s possible to map all the reads with the reference genome.</p>
<div id="remove-barcode" class="section level3">
<h3 class="hasAnchor">
<a href="#remove-barcode" class="anchor"></a>Remove barcode</h3>
<p>Second step, remove barcode. In this data, N = 10.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">args</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span><span class="op">(</span>sysma<span class="op">=</span><span class="st">"param/trim.param"</span>, mytargets<span class="op">=</span><span class="st">"targets_clip.txt"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span></code></pre></div>
<div id="reads-length-distribution" class="section level4">
<h4 class="hasAnchor">
<a href="#reads-length-distribution" class="anchor"></a>Reads Length Distribution</h4>
<p>Read length distribution after barcode removal.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="st">"seeDistribution.R"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span><span class="op">(</span><span class="st">"results/Distribution_barcodeRemoval.pdf"</span><span class="op">)</span>
<span class="fu">seeDistribution</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outfile1</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span>, file<span class="op">=</span><span class="st">"fastq"</span>, batchsize <span class="op">=</span> <span class="fl">100000</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="read-mapping-with-hisat2" class="section level3">
<h3 class="hasAnchor">
<a href="#read-mapping-with-hisat2" class="anchor"></a>Read mapping with <code>HISAT2</code>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">args</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span><span class="op">(</span>sysma<span class="op">=</span><span class="st">"param/hisat2.param"</span>, mytargets<span class="op">=</span><span class="st">"targets_clip_trim.txt"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">sysargs</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="co"># Command-line parameters for first FASTQ file</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/moduleload.html">moduleload</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">modules</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span><span class="op">)</span>
<span class="co">#system("hisat2-build ./data/dataH/hg19.fa ./data/dataH/hg19.fa")</span>
<span class="va">resources</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>walltime<span class="op">=</span><span class="fl">1440</span>, ntasks<span class="op">=</span><span class="fl">1</span>, ncpus<span class="op">=</span><span class="fl">12</span>, memory<span class="op">=</span><span class="fl">10240</span>, email<span class="op">=</span><span class="st">"danielac@ucr.edu"</span><span class="op">)</span>
<span class="va">hisat2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/clusterRun.html">clusterRun</a></span><span class="op">(</span><span class="va">args</span>, conffile<span class="op">=</span><span class="st">".batchtools.conf.R"</span>, template<span class="op">=</span><span class="st">"batchtools.slurm.tmpl"</span>, runid<span class="op">=</span><span class="st">"01"</span>, Njobs<span class="op">=</span><span class="fl">12</span>, resourceList<span class="op">=</span><span class="va">resources</span><span class="op">)</span>
<span class="fu">getStatus</span><span class="op">(</span>reg<span class="op">=</span><span class="va">hisat2</span><span class="op">)</span></code></pre></div>
<p>Check whether all BAM files have been created</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outpaths</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Check all reads mapping results.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">read_statsDF_READS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/alignStats.html">alignStats</a></span><span class="op">(</span>args<span class="op">=</span><span class="va">args</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">read_statsDF_READS</span>, <span class="st">"results/alignStats_READS.xls"</span>, row.names<span class="op">=</span><span class="cn">FALSE</span>, quote<span class="op">=</span><span class="cn">FALSE</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="mapping-unique-reads" class="section level2">
<h2 class="hasAnchor">
<a href="#mapping-unique-reads" class="anchor"></a>Mapping Unique reads</h2>
<div id="collapsing-identical-sequences" class="section level3">
<h3 class="hasAnchor">
<a href="#collapsing-identical-sequences" class="anchor"></a>Collapsing identical sequences</h3>
<p>Collapsing identical sequences in a FASTQ/A file into a single sequence (while maintaining reads counts).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">args</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span><span class="op">(</span>sysma<span class="op">=</span><span class="st">"param/trim_collapser.param"</span>, mytargets<span class="op">=</span><span class="st">"targets_clip.txt"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">sysargs</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span><span class="op">(</span><span class="va">args</span>, make_bam<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">args</span>, file<span class="op">=</span><span class="st">"targets_clip_collapser.txt"</span>, overwrite<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="reads-length-distribution-1" class="section level4">
<h4 class="hasAnchor">
<a href="#reads-length-distribution-1" class="anchor"></a>Reads Length Distribution</h4>
<p>Read length distribution after collapsing the identical reads.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="st">"seeDistribution.R"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span><span class="op">(</span><span class="st">"results/Distribution_collapsing.pdf"</span><span class="op">)</span>
<span class="fu">seeDistribution</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outfile1</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span>, file<span class="op">=</span><span class="st">"fasta"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="removing-barcode" class="section level3">
<h3 class="hasAnchor">
<a href="#removing-barcode" class="anchor"></a>Removing Barcode</h3>
<p>Remove barcode after collapse.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">args</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span><span class="op">(</span>sysma<span class="op">=</span><span class="st">"param/trim_fasta.param"</span>, mytargets<span class="op">=</span><span class="st">"targets_clip_collapser.txt"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span><span class="op">(</span><span class="va">args</span>, make_bam<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">args</span>, file<span class="op">=</span><span class="st">"targets_clip_collap_trim.txt"</span>, overwrite<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div id="reads-length-distribution-2" class="section level4">
<h4 class="hasAnchor">
<a href="#reads-length-distribution-2" class="anchor"></a>Reads Length Distribution</h4>
<p>Read length distribution after barcode removal.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="st">"seeDistribution.R"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span><span class="op">(</span><span class="st">"results/Distribution_Collapsing_barcodeRemoval.pdf"</span><span class="op">)</span>
<span class="fu">seeDistribution</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outfile1</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span>, file<span class="op">=</span><span class="st">"fasta"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="read-mapping-with-hisat2-1" class="section level3">
<h3 class="hasAnchor">
<a href="#read-mapping-with-hisat2-1" class="anchor"></a>Read mapping with <code>HISAT2</code>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">args</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span><span class="op">(</span>sysma<span class="op">=</span><span class="st">"param/hisat2_unique.param"</span>, mytargets<span class="op">=</span><span class="st">"targets_clip_collap_trim.txt"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">sysargs</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="co"># Command-line parameters for first FASTQ file</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/moduleload.html">moduleload</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">modules</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span><span class="op">)</span>
<span class="co">#system("hisat2-build ./data/dataH/hg19.fa ./data/dataH/hg19.fa")</span>
<span class="va">resources</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>walltime<span class="op">=</span><span class="fl">1440</span>, ntasks<span class="op">=</span><span class="fl">1</span>, ncpus<span class="op">=</span><span class="fl">12</span>, memory<span class="op">=</span><span class="fl">10240</span>, email<span class="op">=</span><span class="st">"danielac@ucr.edu"</span><span class="op">)</span>
<span class="va">hisat2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/clusterRun.html">clusterRun</a></span><span class="op">(</span><span class="va">args</span>, conffile<span class="op">=</span><span class="st">".batchtools.conf.R"</span>, template<span class="op">=</span><span class="st">"batchtools.slurm.tmpl"</span>, runid<span class="op">=</span><span class="st">"01"</span>, Njobs<span class="op">=</span><span class="fl">12</span>, resourceList<span class="op">=</span><span class="va">resources</span><span class="op">)</span>
<span class="fu">getStatus</span><span class="op">(</span>reg<span class="op">=</span><span class="va">hisat2</span><span class="op">)</span></code></pre></div>
<p>Check whether all BAM files have been created</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outpaths</a></span><span class="op">(</span><span class="va">args</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Check all reads mapping results.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">read_statsDF_READS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/alignStats.html">alignStats</a></span><span class="op">(</span>args<span class="op">=</span><span class="va">args</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">read_statsDF_READS</span>, <span class="st">"results/alignStats_READS_unique.xls"</span>, row.names<span class="op">=</span><span class="cn">FALSE</span>, quote<span class="op">=</span><span class="cn">FALSE</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
<div id="version-information" class="section level1">
<h1 class="hasAnchor">
<a href="#version-information" class="anchor"></a>Version Information</h1>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R Under development (unstable) (2021-02-04 r79940)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /home/dcassol/src/R-devel/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
##  [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
## [10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] systemPipeR_1.25.11         ShortRead_1.49.2            GenomicAlignments_1.27.2   
##  [4] SummarizedExperiment_1.21.3 Biobase_2.51.0              MatrixGenerics_1.3.1       
##  [7] matrixStats_0.58.0          BiocParallel_1.25.5         Rsamtools_2.7.2            
## [10] Biostrings_2.59.2           XVector_0.31.1              GenomicRanges_1.43.4       
## [13] GenomeInfoDb_1.27.11        IRanges_2.25.9              S4Vectors_0.29.15          
## [16] BiocGenerics_0.37.2         BiocStyle_2.19.2           
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_2.0-0         rjson_0.2.20             hwriter_1.3.2           
##   [4] ellipsis_0.3.1           rprojroot_2.0.2          fs_1.5.0                
##   [7] bit64_4.0.5              AnnotationDbi_1.53.1     fansi_0.4.2             
##  [10] codetools_0.2-18         cachem_1.0.4             knitr_1.32              
##  [13] jsonlite_1.7.2           annotate_1.69.2          dbplyr_2.1.1            
##  [16] png_0.1-7                BiocManager_1.30.12      compiler_4.1.0          
##  [19] httr_1.4.2               backports_1.2.1          assertthat_0.2.1        
##  [22] Matrix_1.3-2             fastmap_1.1.0            limma_3.47.12           
##  [25] htmltools_0.5.1.1        prettyunits_1.1.1        tools_4.1.0             
##  [28] gtable_0.3.0             glue_1.4.2               GenomeInfoDbData_1.2.4  
##  [31] dplyr_1.0.5              rsvg_2.1.1               batchtools_0.9.15       
##  [34] rappdirs_0.3.3           V8_3.4.0                 Rcpp_1.0.6              
##  [37] jquerylib_0.1.3          pkgdown_1.6.1            vctrs_0.3.7             
##  [40] rtracklayer_1.51.5       xfun_0.22                stringr_1.4.0           
##  [43] lifecycle_1.0.0          restfulr_0.0.13          XML_3.99-0.6            
##  [46] edgeR_3.33.3             scales_1.1.1             zlibbioc_1.37.0         
##  [49] BSgenome_1.59.2          VariantAnnotation_1.37.1 ragg_1.1.2              
##  [52] hms_1.0.0                RColorBrewer_1.1-2       yaml_2.2.1              
##  [55] curl_4.3                 memoise_2.0.0            ggplot2_3.3.3           
##  [58] sass_0.3.1               biomaRt_2.47.7           latticeExtra_0.6-29     
##  [61] stringi_1.5.3            RSQLite_2.2.7            BiocIO_1.1.2            
##  [64] desc_1.3.0               checkmate_2.0.0          GenomicFeatures_1.43.8  
##  [67] filelock_1.0.2           DOT_0.1                  rlang_0.4.10            
##  [70] pkgconfig_2.0.3          systemfonts_1.0.1        bitops_1.0-6            
##  [73] evaluate_0.14            lattice_0.20-41          purrr_0.3.4             
##  [76] bit_4.0.4                tidyselect_1.1.0         magrittr_2.0.1          
##  [79] bookdown_0.22            R6_2.5.0                 generics_0.1.0          
##  [82] base64url_1.4            DelayedArray_0.17.10     DBI_1.1.1               
##  [85] pillar_1.6.0             withr_2.4.2              KEGGREST_1.31.1         
##  [88] RCurl_1.98-1.3           tibble_3.1.1             crayon_1.4.1            
##  [91] utf8_1.2.1               BiocFileCache_1.99.6     rmarkdown_2.7           
##  [94] jpeg_0.1-8.1             progress_1.2.2           locfit_1.5-9.4          
##  [97] grid_4.1.0               data.table_1.14.0        blob_1.2.1              
## [100] digest_0.6.27            xtable_1.8-4             brew_1.0-6              
## [103] textshaping_0.3.3        munsell_0.5.0            bslib_0.2.4</code></pre>
</div>
<div id="funding" class="section level1">
<h1 class="hasAnchor">
<a href="#funding" class="anchor"></a>Funding</h1>
<p>This project was supported by funds from the National Science Foundation (NSF).</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-H_Backman2016-bt" class="csl-entry">
H Backman, Tyler W, and Thomas Girke. 2016. <span>“<span class="nocase">systemPipeR: NGS workflow and report generation environment</span>.”</span> <em>BMC Bioinformatics</em> 17 (1): 388. <a href="https://doi.org/10.1186/s12859-016-1241-0">https://doi.org/10.1186/s12859-016-1241-0</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.dcassol.com/">Daniela Cassol</a>, <a href="http://girke.bioinformatics.ucr.edu/">Thomas Girke</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
